resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStepSuite.scala

```diff
@@ -117,6 +117,33 @@ class BasicDriverFeatureStepSuite extends SparkFunSuite {
     assert(featureStep.getAdditionalPodSystemProperties() === expectedSparkConf)
   }
 
+  test("Check driver pod respects kubernetes driver request cores") {
+    val sparkConf = new SparkConf()
+      .set(KUBERNETES_DRIVER_POD_NAME, "spark-driver-pod")
+      .set(CONTAINER_IMAGE, "spark-driver:latest")
+
+    val basePod = SparkPod.initialPod()
+    val requests1 = new BasicDriverFeatureStep(KubernetesTestConf.createDriverConf(sparkConf))
+      .configurePod(basePod)
+      .container.getResources
+      .getRequests.asScala
+    assert(requests1("cpu").getAmount === "1")
```

Comment:
Can we avoid this assumption? You had better get the default value of `DRIVER_CORES` and compare with that.

Reply:
@arunmahadevan . What I meant was this constant assertion, `=== "1"`. If we change the default value of that configuration, this test will fails. We had better get and use the default value from the conf.

Reply:
@dongjoon-hyun , the logic in BasicDriverFeatureStep
`private val driverCpuCores = conf.get(DRIVER_CORES.key, "1")`
, sets the value of `driverCpuCores` to "1" if `spark.driver.cores` is not set in the spark conf. i.e. we don't use the `DRIVER_CORES.defaultValue` there. Let me know if my understanding is correct. 
If so I cannot do something like `assert(requests1("cpu").getAmount === DRIVER_CORES.defaultValue)` here (assume this is what you are suggesting).

Reply:
That should be the following because we should not have a magic number in the code.
```scala
-  private val driverCpuCores = conf.get(DRIVER_CORES.key, "1")
+  private val driverCpuCores = conf.get(DRIVER_CORES)
```

```scala
     val driverCpuQuantity = new QuantityBuilder(false)
-      .withAmount(driverCpuCores)
+      .withAmount(driverCpuCores.toString)
```

Please fix like the above. And don't use magic numbers.

Reply:
updated.